repose-aggregator/components/filters/herp/src/test/scala/org/openrepose/filters/herp/HerpFilterTest.scala

```diff
@@ -293,65 +296,209 @@ class HerpFilterTest extends FunSpec with BeforeAndAfterAll with BeforeAndAfter
+          logEventsPost.size shouldBe condition._2
+        }
+        it("when there are fields with String array values.") {
+          // given:
+          val test = "---foo---"
+          servletRequest.addHeader("X-Roles", test)
+          val matcher = new Match
+          matcher.setField("roles")
+          matcher.setRegex(condition._1)
+          val filterOut = new FilterOut
+          filterOut.getMatch.add(matcher)
+          herpConfig.getFilterOut.add(filterOut)
+
+          // when:
+          herpFilter.configurationUpdated(herpConfig)
+          herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+          // then:
+          def logEventsPre = listAppenderPre.getEvents
+          logEventsPre.size shouldBe 1
+          logEventsPre.get(0).getMessage.getFormattedMessage should include(test)
+
+          def logEventsPost = listAppenderPost.getEvents
+          logEventsPost.size shouldBe condition._2
+        }
+        it("when there are fields with maps with String keys and String array values and the condition is a value.") {
+          // given:
+          val test = "---foo---"
+          servletRequest.setAttribute("http://openrepose.org/queryParams", Map(
+            "---bar---" -> Array("A", "B", "C"),
+            "---buz---" -> Array("1", "2", test)).asJava
+          )
+
+          val matcher = new Match
+          matcher.setField("parameters.---buz---")
+          matcher.setRegex(condition._1)
+          val filterOut = new FilterOut
+          filterOut.getMatch.add(matcher)
+          herpConfig.getFilterOut.add(filterOut)
+
+          // when:
+          herpFilter.configurationUpdated(herpConfig)
+          herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+          // then:
+          def logEventsPre = listAppenderPre.getEvents
+          logEventsPre.size shouldBe 1
+          logEventsPre.get(0).getMessage.getFormattedMessage should include(test)
+
+          def logEventsPost = listAppenderPost.getEvents
+          logEventsPost.size shouldBe condition._2
+        }
+        it("when the matches of a filterOut are AND'd and the filterOut's are OR'd.") {
+          // given:
+          val testOne = "---foo---"
+          val testTwo = "---BUZ---"
+          servletRequest.addHeader("X-User-Name", testOne)
+          servletRequest.addHeader("X-Roles", testTwo)
+          val filtersOut = herpConfig.getFilterOut
+          val filterOne = new FilterOut
+          val matchersOne = filterOne.getMatch
+          val matcherOne = new Match
+          matcherOne.setField("userName")
+          matcherOne.setRegex(condition._1) // Conditionally matches
+          matchersOne.add(matcherOne)
+          val matcherTwo = new Match        // AND'd
+          matcherTwo.setField("roles")
+          matcherTwo.setRegex(".*BUZ.*")    // Always matches
+          matchersOne.add(matcherTwo)
+          filtersOut.add(filterOne)
+          val filterTwo = new FilterOut     // OR'd
```

Comment:
This doesn't test the match criteria in both positions, true, but due to the nature of how Scala filter'ing works it shouldn't matter. Just to avoid any potential issues if the data structure or filter mechanism change, I updated the test to run it through in both positions.